crates/fuel-core/src/service/genesis/importer.rs

```diff
@@ -197,19 +186,15 @@ impl SnapshotImporter {
         let groups = self.snapshot_reader.read::<TableInSnapshot>()?;
         let num_groups = groups.len();
 
-        if num_groups == 0 {
```

Comment:
It seemed like a premature optimization. Genesis can last for hours but a useless task of a few millis cannot be tolerated. 

Now I understand that there are tests that run this a lot and you'd like to optimize for that. I'll add a comment so that this isn't removed again in the future.